src/Perspex.Controls/Primitives/Popup.cs

```diff
@@ -273,9 +273,30 @@ private Point GetPosition()
                     case PlacementMode.Bottom:
                         point = new Point(0, target.Bounds.Height);
                         break;
+
                     case PlacementMode.Right:
                         point = new Point(target.Bounds.Width, 0);
                         break;
+
+                    case PlacementMode.Center:
+
+                        Size childSize;
+
+                        if ((Child != null) && (_topLevel != null))
+                        {
+                            Child.Measure(_topLevel.ClientSize);
```

Comment:
This doesn't look right to me - `_toplevel` is the `Window` etc that hosts the `Popup` so it doesn't make sense that you're measuring the popped-up `Child` using the parent window's `ClientSize`. 

I think it would make more sense to call `PopupRoot.LayoutManager.ExecuteLayoutPass()` here and get the desired size of `PopupRoot` instead. That way it also takes into account `PopupRoot`'s template and any size constraints present on the popup.

If that works - I've not tried it ;)